saleor/payment/gateways/stripe/__init__.py

```diff
@@ -0,0 +1,233 @@
+    client, error = _get_client(**connection_params), ''
+
+    # Get amount from argument or payment
+    amount = amount or payment.total
+
+    try:
+        # Charge without pre-authorize
+        response = _create_stripe_charge(
+            client=client, payment=payment, amount=amount,
+            payment_token=payment_token, capture=True)
+    except stripe.error.StripeError as exc:
+        response = _get_error_response_from_exc(exc)
+        error = exc.user_message
+
+    # Create transaction
+    txn = _create_transaction(
+        payment=payment,
+        amount=amount,
+        kind=TransactionKind.CHARGE,
+        response=response)
+
+    return txn, error
+
+
+def refund(payment, amount, **connection_params):
+    client, error = _get_client(**connection_params), ''
+
+    # Get amount from argument or payment, and convert to stripe's amount
+    amount = amount or payment.total
+    stripe_amount = get_amount_for_stripe(amount, payment.currency)
+
+    capture_txn = payment.transactions.filter(
+        kind=TransactionKind.CHARGE, is_success=True).first()
+
+    if capture_txn is not None:
+        try:
+            # Retrieve stripe charge and refund specific amount
+            stripe_charge = client.Charge.retrieve(capture_txn.token)
+            response = client.Refund.create(
+                charge=stripe_charge.id, amount=stripe_amount)
+        except stripe.error.StripeError as exc:
+            response = _get_error_response_from_exc(exc)
+            error = exc.user_message
+    else:
+        error = errors.ORDER_NOT_CHARGED
+        response = dict()
+
+    # Create transaction
+    txn = _create_transaction(
+        payment=payment,
+        amount=amount,
+        kind=TransactionKind.REFUND,
+        response=response)
+
+    return txn, error
+
+
+def void(payment, **connection_params):
+    client, error = _get_client(**connection_params), ''
+
+    capture_txn = payment.transactions.filter(
+        kind=TransactionKind.CHARGE, is_success=True).first()
+
+    if capture_txn is not None:
+        try:
+            # Retrieve stripe charge and refund all
+            stripe_charge = client.Charge.retrieve(capture_txn.token)
+            response = client.Refund.create(charge=stripe_charge.id)
+        except stripe.error.StripeError as exc:
+            response = _get_error_response_from_exc(exc)
+            error = exc.user_message
+    else:
+        error = errors.ORDER_NOT_CHARGED
+        response = dict()
+
+    # Create transaction
+    txn = _create_transaction(
+        payment=payment,
+        amount=payment.total,
+        kind=TransactionKind.VOID,
+        response=response)
+
+    return txn, error
+
+
+def get_form_class():
+    return StripePaymentModalForm
+
+
+def _get_client(**connection_params):
+    stripe.api_key = connection_params.get('secret_key')
+    return stripe
+
+
+def _create_stripe_charge(client, payment, amount, payment_token, capture):
```

Comment:
In `payment.gateways.stripe`, could you define the private functions at the top of the file? It's most of the time the convention (not sure about Python), but mostly because when reading `authorize`, `capture`, etc. we call private functions, and we expect to have read them already or else, to find them a bit above (or in the imports).

Reply:
Thanks for review, and thanks for your implementation on razorpay, from which I've extracted some patterns and used in stripe. :smile:

Reply:
As the the issue, for c/c++, these functions should definitely be placed at the top. But for python, IMHO it is only a matter of personal choice. There is no standard, and we only need to stick to a specific style.

As for me, I prefer these private functions at the bottom since I actually only would like to expose the public APIs. The readers don't need to know internal implementation (such as these private methods). If they really would like to examine the implementation, scroll down or IDE navigation would do that.

And for stripe, I didn't see any explicit stardand for that. So I would like to still keep this.